spec/brainstem/presenter_spec.rb

```diff
@@ -404,41 +404,73 @@ def workspace.synthetic_id
     end
 
     describe "preloading" do
-      it "preloads associations when they are full model-level associations" do
-        mock(Brainstem::Presenter).ar_preload(anything, anything) do |models, args|
-          expect(args).to eq %w[tasks user]
+
+      #
+      # We have three strategies for introspecting what the AR Preloader
+      # receives:
+      #
+      # 1. Actually looking at what AR receives.
+      #
+      #    This is sub-optimal because AR works differently between versions
+      #    3 and 4, so introspecting is difficult.
+      #
+      # 2. Looking at what the proc that encapsulates the AR methods is
+      #    called with.
+      #
+      #    This is difficult to do because we don't control the instantiation
+      #    of the Preloader, which makes injecting this hard.
+      #
+      # 3. Intercept the instantiating parent function and append the
+      #    inspectable proc.
+      #
+      #    This is probably the grossest of all the above options, and I
+      #    suspect it's the greatest indication that we're testing
+      #    inappropriately here -- a fact that I think bears weight given
+      #    we're making unit-level assertions in an integration spec.
+      #    However, there's also some value in asserting that these are
+      #    passed through without digging into Rails internals. However,
+      #    further 'purity' improvements could be made by introspecting on
+      #    AR's actual data structures.
+      #
+      #    That's about three shades on the side of overkill, though.
+      #
+      def preloader_should_receive(hsh)
```

Comment:
I've moved `ar_preload` into the Preloader as the presenter shouldn't care about the actual method of preloading. 

Inside `Preloader`, it's appropriate to inject a method or a proc as a collaborator because it should be insulated from the actual implementation within ActiveRecord—or, put another way, what the preloader needs is to send a message to ActiveRecord, but just *which* message is actually an implementation detail of AR and should be injected as a dependency. 

We could move the `ar_preload` method entirely within the Preloader, but that closely binds Preloader to AR. In some situations (testing being the biggest example), we want to inject a different collaborator that handles the way that preloading is done.